test/functional/gridfs_stream.test.js

```diff
@@ -1166,13 +1008,13 @@ describe('GridFS Stream', function () {
                 });
 
                 download.on('end', function () {
-                  var result = testSpec.assert.result;
+                  const result = testSpec.assert.result;
                   if (!result) {
                     if (errorReported) {
                       return;
                     }
 
-                    test.ok(false);
+                    expect.fail('errorReported should be set'); // ??
```

Comment:
I don't entirely know what's happening here either, but it looks like an attempt at clean up, which won't execute if we actually throw an error above, so I think we actually need to move this expect to below the client.close, and not pass the done to client.close?